src/redux/search.js

```diff
@@ -0,0 +1,92 @@
+import { Observable } from 'rxjs/Observable';
+import queryString from 'query-string';
+import 'rxjs/add/observable/from';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/pluck';
+import 'rxjs/add/operator/catch';
+
+// action types
+const EHOLDINGS_SEARCH = 'EHOLDINGS_SEARCH';
+const EHOLDINGS_SEARCH_RESOLVE = 'EHOLDINGS_SEARCH_RESOLVE';
+const EHOLDINGS_SEARCH_ERRORED = 'EHOLDINGS_SEARCH_ERRORED';
+
+// actions
+export const searchHoldings = (searchType, query) => ({
+  type: EHOLDINGS_SEARCH,
+  searchType,
+  query
+});
+
+// reducer creator
+export function createSearchReducer(searchType) {
+  const initialState = {
+    query: {},
+    isLoading: false,
+    isErrored: false,
+    records: [],
+    error: null
+  };
+
+  return (state = initialState, action) => {
+    if (searchType === action.searchType) {
+      switch (action.type) {
+        case EHOLDINGS_SEARCH:
+          return {
+            ...state,
+            query: action.query,
+            isLoading: true,
+            isErrored: false,
+            error: null
+          };
+        case EHOLDINGS_SEARCH_RESOLVE:
+          return {
+            ...state,
+            isLoading: false,
+            records: action.payload
+          };
+        case EHOLDINGS_SEARCH_ERRORED:
+          return {
+            ...state,
+            isLoading: false,
+            isErrored: true,
+            error: action.error
+          };
+        default:
+          return state;
+      }
+    } else {
+      return state;
+    }
+  };
+}
+
+// epic creator
+export function createSearchEpic(searchType, {
+  endpoint = '',
+  defaultParams = {},
+  recordsKey = searchType
+} = {}) {
+  const getSearchURL = (baseURL, params) => {
+    const search = queryString.stringify({
+      ...defaultParams,
+      ...params
+    });
+
+    return `${baseURL}/${endpoint}?${search}`;
+  };
+
+  return (action$, { getState }) => (
+    action$.ofType(EHOLDINGS_SEARCH)
+      .switchMap((action) => {
+        const { okapi } = getState();
+
+        const request = fetch(getSearchURL(okapi.url, action.query), {
+          headers: { 'X-Okapi-Tenant': okapi.tenant }
```

Comment:
we do this a lot.  i wonder if there's a way to apply the okapi header (soon to be headers) wholesale...

Reply:
We could do something like define a helper such as `fetchFromOkapi(okapiObject, resourceEndpoint, fetchOptions)` and construct the headers there